applications/GeoMechanicsApplication/custom_elements/small_strain_U_Pw_diff_order_element.cpp

```diff
@@ -712,7 +712,7 @@ void SmallStrainUPwDiffOrderElement::AssignPressureToIntermediateNodes()
         ThreadSafeNodeWrite(rGeom[16], WATER_PRESSURE, 0.5 * (p4 + p5));
         ThreadSafeNodeWrite(rGeom[17], WATER_PRESSURE, 0.5 * (p5 + p6));
         ThreadSafeNodeWrite(rGeom[18], WATER_PRESSURE, 0.5 * (p6 + p7));
-        ThreadSafeNodeWrite(rGeom[19], WATER_PRESSURE, 0.5 * (p7 + p0));
+        ThreadSafeNodeWrite(rGeom[19], WATER_PRESSURE, 0.5 * (p7 + p4));
```

Comment:
@ipouplana For your information, we found an issue with the computation of the fluid pressure in our diff order element. Since it looks very similar to the code in `small_strain_U_Pl_diff_order_element.cpp`, we think the same bug is present there (the average is taken here of p7 and p0, while it should be p7 and p4):

https://github.com/KratosMultiphysics/Kratos/blob/4b16974076baf600714700406b214774c2901e88/applications/PoromechanicsApplication/custom_elements/one-phase_flow/small_strain_U_Pl_diff_order_element.cpp#L566

When looking at the node ordering in Hexahedra3D20, this becomes clear:
```
/**
 * @class Hexahedra3D20
 * @ingroup KratosCore
 * @brief A twenty node hexahedra geometry with serendipity shape functions
 * @details The node ordering corresponds with:
 *      3----10----2
 *      |\         |\
 *      | 15       | 14
 *     11  \       9  \
 *      |   7----18+---6
 *      |   |      |   |
 *      0---+-8----1   |
 *       \  19      \  17
 *       12 |        13|
 *         \|         \|
 *          4----16----5
 * @author Riccardo Rossi
 * @author Janosch Stascheit
 * @author Felix Nagel
 */
```

Reply:
Thank you @rfaasse for pointing this out. I’ll correct it in another PR.